### PR TITLE
[pump] new definitions, v1.0

### PR DESCRIPTION
**Pull request by tlaziuk**
- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: pump/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+// Project: https://github.com/mafintosh/pump
+// Definitions by: Tomek Łaziuk <https://github.com/tlaziuk>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+declare function pump(streams: pump.Stream[], callback?: pump.Callback): pump.Stream[];
+
+// callback have to be passed as last argument
+declare function pump(...streams: Array<pump.Stream | pump.Callback>): pump.Stream[];
```
Comment (aozgaa) [inline pump/index.d.ts]: Can this be typed as
``` ts
declare function pump(...streams: Array<pump.Stream | pump.Callback>, lastCallBack: pump.Callback): pump.Stream[];
```
?

File: pump/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Type definitions for pump 1.0
+// Project: https://github.com/mafintosh/pump
+// Definitions by: Tomek Łaziuk <https://github.com/tlaziuk>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+declare function pump(streams: pump.Stream[], callback?: pump.Callback): pump.Stream[];
```
Comment (aozgaa) [inline pump/index.d.ts]: Can you add a test which doesn't call `pump` with a callback?
Can you add a test which passes an array of streams?

File: pump/index.d.ts (hunk 1 above)
Comment (tlaziuk) [inline pump/index.d.ts]: no, it would be not valid TypeScript code:
`pump/index.d.ts|11 col 23 error| A rest parameter must be last in a parameter list.`

Comment (tlaziuk) [inline pump/index.d.ts]: actually maybe it is possible, but I don't know how to achieve that, that's very rare to put rest parameters at first

I've been looking for something like typed array with index -1 type, but such thing doesn't exists

File: pump/index.d.ts (hunk 2 above)
Comment (tlaziuk) [inline pump/index.d.ts]: done

File: pump/index.d.ts (hunk 1 above)
Comment (aozgaa) [inline pump/index.d.ts]: Right :). Can you make the comment above a doc-comment then, so at least consumers of the interface get some warning about the ordering of parameters?

**Pull request merged**